### DISCUSSION TITLE
[FR] Add an Event to register "Compatible Field Types" (to hide the yellow triangle)

### DISCUSSION THREAD

**Discussion by lindseydiloreto** [craftcms/cms#13793] [Ideas] [answered]
I have two 3rd-party plugins, both of which use 3rd-party database tables. Neither of them use the default `content` table.

I would like to programmatically declare that these two field types are compatible (and hide the yellow triangle)...

<img src="https://github.com/craftcms/cms/assets/5309692/87bca03f-eaa6-48b5-92f2-8a183faa4980" width="212">

<hr>

Would it be possible to add an Event (or something else), which might allow me to declare that these field types are compatible?

**Comment by brandonkelly** [comment, marked as answer]
Added an event for the next release! (ca9b40adf6a748c1a910898b3283f9bb6b66bf59)

```php
use craft\base\Event;
use craft\events\DefineCompatibleFieldTypesEvent;
use craft\services\Fields;

Event::on(
    Fields::class, 
    Fields::EVENT_DEFINE_COMPATIBLE_FIELD_TYPES,
    function (DefineCompatibleFieldTypesEvent $event) {
        if ($event->field instanceof GoogleMapsAddress) {
            $event->compatibleTypes[] = MapboxAddress::class;
        }
    }
);
```

**Reply by lindseydiloreto** [reply]
Amazing, thanks @brandonkelly!! 🍺 👏

**Reply by brandonkelly** [reply]
Craft 4.5.7 is out with that new event!

**Reply by lindseydiloreto** [reply]
Works like a charm, thanks Brandon! 🎉

**Reply by lindseydiloreto** [reply]
Rolled into both plugins and working great... 👍 

- https://github.com/doublesecretagency/craft-mapbox/releases/tag/1.1.2
- https://github.com/doublesecretagency/craft-googlemaps/releases/tag/4.3.8

<img src="https://github.com/craftcms/cms/assets/5309692/b4a8a6c4-4ce7-469d-8d22-75f707f6ad7d" width="210">

**Reply by brandonkelly** [reply]
Good to hear!
